gio/src/list_store.rs

```diff
@@ -100,4 +104,17 @@ mod tests {
         assert_eq!(list.item(0), Some(item0.upcast()));
         assert_eq!(list.item(1), Some(item1.upcast()));
     }
+
+    #[test]
+    fn extend() {
+        let item0 = ListStore::new(ListStore::static_type());
+        let item1 = ListStore::new(ListStore::static_type());
+        let mut list = ListStore::new(ListStore::static_type());
+        list.extend(&[&item0, &item1]);
+        assert_eq!(list.item(0).as_ref(), Some(item0.upcast_ref()));
```

Comment:
```suggestion
        assert_eq!(list.item(0), Some(item0.upcast()));
```

Reply:
No, because `upcast()` consumes the value and then it's gone and can't be used for the following comparison :)

Reply:
Ups, sorry. 

I don't have an "approve" button, but i approve this PR.